From: jingi_jeong <jingi_jeong@tmax.co.kr>
Date: Tue, 20 Apr 2021 17:03:36 +0900
Subject: [feat] search : Preventing Text Input In Search Box While Pressing
 Specific Modifier

- To prevent search operation while the modifier other than shift and control is pressed

[IMS] N/A
---
 extensions/search/baseMenuLayout.js | 7 +++++++
 extensions/search/menuWidgets.js    | 7 +++++++
 2 files changed, 14 insertions(+)

diff --git a/extensions/search/baseMenuLayout.js b/extensions/search/baseMenuLayout.js
index 5a726dc..05834ae 100644
--- a/extensions/search/baseMenuLayout.js
+++ b/extensions/search/baseMenuLayout.js
@@ -172,6 +172,13 @@ var BaseLayout = class { // eslint-disable-line no-unused-vars
             return Clutter.EVENT_PROPAGATE;
         }
 
+        // To prevent search while pressing modifier other than shift
+        let state = event.get_state();
+        if (state) {
+            if (state & ~Clutter.ModifierType.SHIFT_MASK)
+                return Clutter.EVENT_STOP;
+        }
+
         let symbol = event.get_key_symbol();
         let key = event.get_key_unicode();
 
diff --git a/extensions/search/menuWidgets.js b/extensions/search/menuWidgets.js
index f185fcb..e4ca2af 100644
--- a/extensions/search/menuWidgets.js
+++ b/extensions/search/menuWidgets.js
@@ -972,6 +972,13 @@ var SearchBox = class ArcMenuSearchBox {
     }
 
     _onKeyPress(actor, event) {
+        // To prevent search while pressing modifier other than control and shift, especially <super>
+        let state = event.get_state();
+        if (state) {
+            if (state & ~(Clutter.ModifierType.SHIFT_MASK | Clutter.ModifierType.CONTROL_MASK))
+                return Clutter.EVENT_STOP;
+        }
+
         this.emit('key-press-event', event);
         return Clutter.EVENT_PROPAGATE;
     }
